src/remind_me/cog.py

```diff
@@ -0,0 +1,123 @@
+from asgiref.sync import sync_to_async
+from discord.ext import commands
+from discord.ext.commands import Bot, Context
+from discord_bot.cog import BaseCog
+from .models import Reminder
+
+
+logger = getLogger(__name__)
+
+
+LIST_REMINDERS_HEADERS = ["ID", "Reminder Time", "Reminder Text"]
+
+
+class RemindMeCog(BaseCog):
+    @commands.Cog.listener()
+    async def on_ready(self) -> None:
+        existing_reminders = Reminder.objects.all()
+        start_time = datetime.datetime.now(datetime.UTC)
+        # Check for any reminders that may have expired while the bot was down and send them
+        # or create timers for the remaining ones
+        async for existing_reminder in existing_reminders:
+            if existing_reminder.reminder_time <= start_time:
+                await self._send_reminder(existing_reminder)
+            else:
+                delta = (existing_reminder.reminder_time - start_time).seconds
+                self._create_timer(delta, existing_reminder)
+
+    def _create_timer(self, delta: float, reminder: Reminder) -> None:
+        logger.info(
+            "Reminding Discord User (ID=%d) in %0.2f seconds in channel %d",
+            reminder.creator_id,
+            delta,
+            reminder.initial_channel_id,
+        )
+        asyncio.get_running_loop().call_later(
+            delta, lambda: asyncio.create_task(self._send_reminder(reminder))
+        )
+
+    async def _send_reminder(self, reminder: Reminder) -> None:
+        channel = self.bot.get_partial_messageable(reminder.initial_channel_id)
+        if not channel:
+            logger.warn(
+                "No channel was found for %d, doing nothing.",
+                reminder.initial_channel_id,
+            )
+            await reminder.adelete()
+        await channel.send(
+            f'<@{reminder.creator_id}>, I am reminding you about "{reminder.reminder_text}"'
+        )
+        await reminder.adelete()
+
+    @commands.command(
+        help="Create a reminder to get pinged about something in the future"
+    )
+    async def remindme(
+        self, ctx: Context[Bot], time_text: str, reminder_text: str
+    ) -> None:
+        settings: dateparser._Settings = {
+            "PREFER_DATES_FROM": "future",
+            "RELATIVE_BASE": datetime.datetime.now(),
+        }
+        reminder_time = dateparser.parse(time_text, settings=settings)
+        if not reminder_time:
+            # We couldn't parse their reminder time so log and let user known
+            logger.warn("User provided an invalid reminder time (%s)", time_text)
+            await ctx.message.reply(
+                "Your reminder time is invalid, please make sure it is correct and try again!"
+            )
+            return
+        reminder = Reminder(
+            creator_id=ctx.author.id,
+            reminder_text=reminder_text,
+            reminder_time=reminder_time,
+            initial_channel_id=ctx.channel.id,
+        )
+        await reminder.asave()
```

Comment:
This is mostly a style thing but I prefer to do this with 'create' statements rather than calling the constructor:

```suggestion
        reminder = await Reminder.objects.acreate(
            creator_id=ctx.author.id,
            reminder_text=reminder_text,
            reminder_time=reminder_time,
            initial_channel_id=ctx.channel.id,
        )
```